source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/AdaptiveCardRenderer.java

```diff
@@ -239,11 +247,18 @@ else if (alignment == ActionAlignment.Center.swigValue())
             viewGroup.addView(actionButtonsLayout);
         }
 
-        for (int i = 0; i < size && i < hostConfig.getActions().getMaxActions(); i++)
+        int i = 0;
+        int maxActions = (int) hostConfig.getActions().getMaxActions();
+        for (; i < size && i < maxActions; i++)
         {
             BaseActionElement actionElement = baseActionElementList.get(i);
             ActionElementRenderer.getInstance().render(renderedCard, context, fragmentManager, actionButtonsLayout, actionElement, cardActionHandler, hostConfig);
         }
+
+        if (i >= maxActions)
```

Comment:
should be i > maxActions

Reply:
no. Should be >= .  In fact, i will never be greater than because the previous for loop will exit once i == maxActions 

---
In reply to: [175174739](https://github.com/Microsoft/AdaptiveCards/pull/1250#discussion_r175174739) [](ancestors = 175174739)

Reply:
Talked offline. Need to also make sure size != maxactions. Good catch!

---
In reply to: [175221222](https://github.com/Microsoft/AdaptiveCards/pull/1250#discussion_r175221222) [](ancestors = 175221222,175174739)